### PR TITLE
Implement DPoS unlocking command - Closes #6726 

### LINKED ISSUE #6726: Implement unlock command for DPoS module
### Description

Implement a token unlocking command in `unlock.ts` in the `commands` folder as in the LIP (https://research.lisk.com/t/define-state-and-state-transitions-of-dpos-module/320#unlock-52)

- Define command id as in the LIP. Command has no params so an empty schema is required.
- Define command name as `unlockToken`
- Implement `execute` hook as in the LIP

### Acceptance Criteria

- Possible scenarios are tested in the unit test for `execute` hook
- All logic exist in the previous DPoS module
- keep in mind #6793

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #6726 

### How was it solved?

Implemented unlock command and test following scenarios:

- [x] When non self-voted non punished account waits 2000 blocks
- [x] When non self-voted non punished account did not wait 2000 blocks
- [x] When self-voted non punished account waits 260000 blocks
- [x] When self-voted non punished account did not wait 260000 blocks
- [x] When non self-voted punished account has waited pomHeight + 260000 and unvoteHeight + 2000 blocks
- [x] When non self-voted punished account has waited pomHeight + 260000 blocks but not waited for unlockHeight + 2000 blocks
- [x] When non self-voted punished account has not waited pomHeight + 260,000 blocks but waited unlockHeight + 2000 blocks
- [x]  When non self-voted punished account has not waited pomHeight + 260,000 blocks but waited unvoteHeight + 2000 blocks and pomHeight is more than unvoteHeight + 2000 blocks
- [x]  When non self-voted punished account has not waited pomHeight + 260000 blocks but waited unvoteHeight + 2000 blocks and pomHeight is equal to unvoteHeight + 2000 blocks
- [x] When self-voted punished account waits 780000 blocks and waits unvoteHeight + 260000 blocks since pomHeight
- [x] When self-voted punished account waits 780000 blocks and but did not wait unvoteHeight + 260000 blocks since pomHeight
- [x] When self-voted punished account has not waited 780000 blocks and waits unvoteHeight + 260000 blocks since pomHeight
- [x] When self-voted punsished account has not waited pomHeight + 260000 blocks but waited unvoteHeight + 2000 blocks and pomHeight is more than unvoteHeight + 2000 blocks

### How was it tested?

Added unit tests

### CODE REVIEW

File: framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+			await delegateSubstore.setWithSchema(
+				delegate2.address,
+				{
+					...defaultDelegateInfo,
+					name: 'punishedvoter2',
+					pomHeights: [lastBlockHeight],
+				},
+				delegateStoreSchema,
+			);
+			// This covers scenario: has not waited pomHeight + 260000 blocks but waited unvoteHeight + 2000 blocks and pomHeight is equal to unvoteHeight + 2000 blocks
```
Comment (shuse2) [inline framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts:251]:
```suggestion
			// This covers scenario: has not waited pomHeight + 260,000 blocks but waited unvoteHeight + 2000 blocks and pomHeight is equal to unvoteHeight + 2000 blocks
```

File: framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+		unlockCommand.addDependencies({
+			tokenIDDPoS: { chainID: 0, localID: 0 },
+			tokenAPI: mockTokenAPI,
+		});
+		db = new InMemoryKVStore() as never;
+		stateStore = new StateStore(db);
+		delegateSubstore = stateStore.getStore(MODULE_ID_DPOS, STORE_PREFIX_DELEGATE);
+		voterSubstore = stateStore.getStore(MODULE_ID_DPOS, STORE_PREFIX_VOTER);
+		lastBlockHeight = 8760000;
+		header = new BlockHeader({
```
Comment (shuse2) [inline framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts:93]: you can use testing `createFakeBlockHeader`

File: framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+		};
+		unlockCommand.addDependencies({
+			tokenIDDPoS: { chainID: 0, localID: 0 },
+			tokenAPI: mockTokenAPI,
+		});
+		db = new InMemoryKVStore() as never;
+		stateStore = new StateStore(db);
+		delegateSubstore = stateStore.getStore(MODULE_ID_DPOS, STORE_PREFIX_DELEGATE);
+		voterSubstore = stateStore.getStore(MODULE_ID_DPOS, STORE_PREFIX_VOTER);
+		lastBlockHeight = 8760000;
```
Comment (shuse2) [inline framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts:92]:
```suggestion
		blockHeight = 8760000;
```
this height is currently being executed

File: framework/src/modules/dpos_v2/utils.ts [hunk 4]
```diff
@@ -119,18 +67,6 @@ export const validateSignature = (
 	bytes: Buffer,
 ): boolean => verifyData(tag, networkIdentifier, bytes, signature, publicKey);
 
-export const isCurrentlyPunished = (height: number, pomHeights: ReadonlyArray<number>): boolean => {
```
Comment (shuse2) [inline framework/src/modules/dpos_v2/utils.ts:122]: this is still used in other function, so it should be kept

File: framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts [hunk 5]
```diff
@@ -0,0 +1,450 @@
...
+	});
+
+	describe(`when self-voted punished account does not wait ${SELF_VOTE_PUNISH_TIME} blocks and waits unvoteHeight + ${WAIT_TIME_SELF_VOTE} blocks since pomHeight`, () => {
+		beforeEach(async () => {
+			await delegateSubstore.setWithSchema(
+				transaction.senderAddress,
+				{
+					...defaultDelegateInfo,
+					name: 'punishedselfvoter',
+					pomHeights: [lastBlockHeight - 1],
```
Comment (shuse2) [inline framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts:409]: just note here if `lastBlockHeight - 1` will make `unvoteHeight` to be within punishment `lastPomHeight < unlockingObject.unvoteHeight + WAIT_TIME_SELF_VOTE` period.

File: framework/src/modules/dpos_v2/utils.ts [hunk 6]
```diff
@@ -147,3 +83,46 @@ export const getVoterOrDefault = async (voterStore: SubStore, address: Buffer) =
...
+export const hasWaited = (
+	unlockingObject: UnlockingObject,
+	senderAddress: Buffer,
+	height: number,
+) => {
+	let delayedAvailability: number;
+
+	// If self-vote
+	if (unlockingObject.delegateAddress.equals(senderAddress)) {
+		delayedAvailability = 260000;
```
Comment (shuse2) [inline framework/src/modules/dpos_v2/utils.ts:96]: Prefer to use constants

**Review by shuse2 — CHANGES_REQUESTED** [review]

**Review by shuse2 — APPROVED** [review]
LGTM! Good job 🚀

**Review by ishantiw — APPROVED** [review]
Great work @mitsuaki-u 👏

**Pull request merged**